utils/hare_cov/hare_coverage.py

```diff
@@ -0,0 +1,362 @@
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from abc import ABC, abstractmethod
+from subprocess import Popen, PIPE
+from typing import Optional, Callable, List, Dict, Type
+from os import walk, sep, path, getcwd
+import logging
+
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s [%(levelname)s] %(message)s',
+                    filename='coverage.log')
+
+
+class FailedToGenerateReport(BaseException):
+    """Used to raise exception while report generation failed."""
+
+    pass
+
+
+class DirectoryScanner:
+    """
+    Scans the directory according to the given pattern. It never return files
+    which are not listed with 'include' keyword argument.
+    """
+
+    def __init__(self,
+                 pattern: str,
+                 callback: Optional[Callable[[str], bool]] = None,
+                 **kwargs):
+        """
+        Sets a pattern for directory named to be search. Sets the filter for
+        the pattern passed with 'omit' keyword argument.
+        :param pattern: Directory name to scan
+        :param callback: callback for comparison
+        :param kwargs: filtering files which are not listed in include kwarg.
+        """
+        self.pattern = pattern
+        self.callback = callback or self.comparator
+        self.inc_kwargs = kwargs.get("include",

```

Comment:
[optional]
Can we put `include` kwarg directly into the `__init__` method signature? I don't think that we buy anything by having generic `**kwargs` in the signature. The end developer may miss this `include` parameter when looking at the signature only.

Reply:
Done